packages/ptokens-assets-evm/src/lib/index.ts

```diff
@@ -44,84 +54,76 @@ export enum EVENT_NAMES {
+        originNetworkId: _obj.originatingNetworkId || _obj.originNetworkId || _obj.networkId,
+        nonce: _obj.nonce,
+        originAccount: _obj.originAccount,
+        destinationAccount: _obj.destinationAccount,
+        destinationNetworkId: _obj.destinationNetworkId,
+        forwardDestinationNetworkId: _obj.forwardDestinationNetworkId,
+        underlyingAssetName: _obj.underlyingAssetName,
+        underlyingAssetSymbol: _obj.underlyingAssetSymbol,
+        underlyingAssetDecimals: _obj.underlyingAssetDecimals,
+        underlyingAssetTokenAddress: _obj.underlyingAssetTokenAddress,
+        underlyingAssetNetworkId: _obj.underlyingAssetNetworkId,
+        assetAmount: _obj.assetAmount,
+        userDataProtocolFeeAssetAmount: _obj.userDataProtocolFeeAssetAmount,
+        networkFeeAssetAmount: _obj.networkFeeAssetAmount,
+        forwardNetworkFeeAssetAmount: _obj.forwardNetworkFeeAssetAmount,
+        userData: _obj.userData || '0x',
+        optionsMask: _obj.optionsMask,
+        isForProtocol: _obj.isForProtocol,
+      },
+    ],
+  )
+
+  return toHex(sha256(toBytes(coded)))
 }
 
-export const getOperationIdFromLog = (_log: Log, _networkId: NetworkId = null) => {
-  const decodedLog = decodeLog(getEventInputsFromSignature(_log.topics[0].toString()), _log.data.toString(), [])
-  return getOperationIdFromObj(
+export const getOperationIdFromLog = (_log: Log<bigint>, _networkId: NetworkId | null = null) => {
+  const decodedLog = decodeEventLog({
+    abi: pNetworkHubAbi,
+    data: _log.data,
+    topics: _log.topics,
+  })
+  const id = getOperationIdFromObj(
     Object.assign(
       {},
-      decodedLog.operation ? decodedLog.operation : decodedLog,
+      'operation' in decodedLog.args ? decodedLog.args.operation : decodedLog.args,
       {
         transactionHash: _log.transactionHash,
         blockHash: _log.blockHash,
       },
       _networkId ? { networkId: _networkId } : {},
     ),
   )
+  return id
 }
 
-export const getOperationIdFromTransactionReceipt = (_networkId: NetworkId, _receipt: TransactionReceipt) => {
-  return getOperationIdFromLog(
-    _receipt.logs.find(
-      (_log) =>
-        _log.topics[0] === eventNameToSignatureMap.get(EVENT_NAMES.USER_OPERATION) ||
-        _log.topics[0] === eventNameToSignatureMap.get(EVENT_NAMES.OPERATION_QUEUED) ||
-        _log.topics[0] === eventNameToSignatureMap.get(EVENT_NAMES.OPERATION_EXECUTED) ||
-        _log.topics[0] === eventNameToSignatureMap.get(EVENT_NAMES.OPERATION_CANCELLED),
-    ),
-    _networkId,
+export const getOperationIdFromTransactionReceipt = (_networkId: NetworkId, _receipt: TransactionReceipt<bigint>) => {
+  const relevantLog = _receipt.logs.find(
+    (_log) =>
+      _log.topics[0] === eventNameToSignatureMap.get(EVENT_NAMES.USER_OPERATION) //||
+      // _log.topics[0] === eventNameToSignatureMap.get(EVENT_NAMES.OPERATION_QUEUED) ||
```

Comment:
The commented lines should work as well, isn't it?

Reply:
Not really, In order to have the new opId I need to search in the interim operatinexecuted only for user operations. In no other part in reality I have ot search for anything else given that the opId is computed only at the user operation.